name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/preferences/BackupsPreferencePage.java

```diff
@@ -0,0 +1,123 @@
+        addField(backupFolderAbsoluteEditor);
+
+        String relativeEditorMessage = MessageFormat.format("Relativer Backup Order (z.B. {0})",
+                        Paths.get(".", "back").toString()); //$NON-NLS-1$ //$NON-NLS-2$
+        backupFolderRelativeEditor = new StringFieldEditor(UIConstants.Preferences.BACKUP_FOLDER_RELATIVE,
+                        relativeEditorMessage, getFieldEditorParent());
+        addField(backupFolderRelativeEditor);
+    }
+
+    protected static ComboFieldEditor createBackupModeSelect(Composite parent)
+    {
+        BACKUP_MODE[] backupModes = BACKUP_MODE.values();
+        String[][] entryNamesAndValues = new String[backupModes.length][2];
+
+        for (int i = 0; i < backupModes.length; i++)
+        {
+            entryNamesAndValues[i][0] = backupModes[i].getTitle();
+            entryNamesAndValues[i][1] = backupModes[i].getPreferenceValue();
+        }
+
+        return new ComboFieldEditor(UIConstants.Preferences.BACKUP_MODE, "Backup mode", entryNamesAndValues, parent);
+    }
+
+    @Override
+    protected void initialize()
+    {
+        super.initialize();
+
+        updateBackupFolderEditors(BACKUP_MODE.byValue(
+                        backupModeEditor.getPreferenceStore().getString(backupModeEditor.getPreferenceName())));
+
+        String relativeFolder = backupFolderRelativeEditor.getStringValue();
+        if (relativeFolder == null || relativeFolder.isBlank())
+        {
+            backupFolderRelativeEditor.setStringValue(ClientInput.DEFAULT_RELATIVE_BACKUP_FOLDER);
+        }
+    }
+
+    @Override
+    public void propertyChange(PropertyChangeEvent event)
+    {
+        super.propertyChange(event);
+
+        if (event.getProperty().equals(FieldEditor.VALUE) && event.getSource().equals(backupModeEditor))
+        {
+            updateBackupFolderEditors(BACKUP_MODE.byValue((String) event.getNewValue()));
+        }
+    }
+
+    private void updateBackupFolderEditors(BACKUP_MODE mode)
+    {
+        if (mode == null)
+        {
+            mode = BACKUP_MODE.getDefault();
+        }
+
+        switch (mode)
+        {
+            case NEXT_TO_FILE:
+                backupFolderAbsoluteEditor.setEnabled(false, getFieldEditorParent());
+                backupFolderRelativeEditor.setEnabled(false, getFieldEditorParent());
+                return;
+            case ABSOLUTE_FOLDER:
+                backupFolderAbsoluteEditor.setEnabled(true, getFieldEditorParent());
+                backupFolderRelativeEditor.setEnabled(false, getFieldEditorParent());
+                return;
+            case RELATIVE_FOLDER:
+                backupFolderAbsoluteEditor.setEnabled(false, getFieldEditorParent());
+                backupFolderRelativeEditor.setEnabled(true, getFieldEditorParent());
+                return;
+            default:
+                throw new RuntimeException("Unsupported backup mode: " + mode.name()); //$NON-NLS-1$

```

Comment:
For those internal errors (which indicate a programming error if they occur), I usually use the IllegalArgumentException (Sonar does not like the generic RuntimeException)